authFunctions/login.go

```diff
@@ -0,0 +1,111 @@
+package authFunctions
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/dgrijalva/jwt-go"
+	_ "github.com/go-sql-driver/mysql"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func createToken(user User) (string, error) {
+	var err error
+
+	// 鍵となる文字列(多分なんでもいい)
+	secret := "auth-using-go"
+
+	// Token を作成
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"email": user.Email,
+		"iss":   "FunCloudAPI", // JWT の発行者が入る(文字列は任意)
+	})
+
+	//Dumpを吐く
+	spew.Dump(token)
+
+	tokenString, err := token.SignedString([]byte(secret))
+
+	fmt.Println("-----------------------------")
+	fmt.Println("tokenString:", tokenString)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return tokenString, nil
+}
+
+func Login(w http.ResponseWriter, r *http.Request) {
+	var user User
+	var error Error
+	var jwt JWT
+
+	json.NewDecoder(r.Body).Decode(&user)
```

Comment:
たぶん処理的に代入式でかけないと思うんだけど
// user := json.NewDecoder(r.Body)
みたいな処理してるのかな？

Reply:
Decodeメソッドはerrorを返すので、受け取って上げましょう
[参考](https://godoc.org/encoding/json#Decoder.Decode)
```suggestion
	err := json.NewDecoder(r.Body).Decode(&user)
	if err != nil {
	    log.Println(err)
	    return //関数を終了する
	}
```

Reply:
わかりました，ありがとうございます！！